app/models/church_value.rb

```diff
@@ -1,3 +1,14 @@
 class ChurchValue < ActiveRecord::Base
   belongs_to :church
+
+  def self.values_for(church_ids, period)
+    key = "#{self.class}/#{period}/#{church_ids.join(',')}"
+    Rails.cache.fetch(key, expires_in: 1.minute) do
```

Comment:
The fact that you're only caching for a minute makes the fancier key about not very important.